packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -134,6 +134,9 @@ function UnforwardedControlWithError< C extends React.ReactElement >(
 			case 'validating': {
 				// Wait before showing a validating state.
 				const timer = setTimeout( () => {
+					validityTarget?.setCustomValidity( '' );
+					setErrorMessage( validityTarget?.validationMessage );
```

Comment:
Maybe in this case we want to `setErrorMessage( undefined )`? There is no error while validation is running.

The use case is interaction of the `onValidate` custom (server-side) validation with client-side validation, e.g., when there is something like `<input type="email">` and the value is not email.

In that case, when additional custom validation is in flight, and status is `validating`, do we want `errorMessage` to show the "not email" error or not.

I'm not sure if we even support this combination of custom and native validation.

Reply:
Turns out, in practice it did feel better to [make the user wait](https://github.com/WordPress/gutenberg/pull/71260#issuecomment-3214306694) in all cases 😅 Going with `setErrorMessage( undefined )` as in your suggestion.